Embarassing solution

tlimato opened the pull request:
I can't believe I never uploaded and submitted this code. I'm so sorry this was just living on my Desktop computer for the past month and a half.
I rewrote the markdown file as it was wrong and incomplete and some of the code was wrong so that was rewritten as well. I apologize for the late submission of such an beginner assignment.

tlimato committed:
solution

I can't believe I never uploaded and submitted this code.

tlimato committed:
Merge branch 'main' into feedback

larskotthoff commented on README.md:
```diff
@@ -1,17 +1,35 @@
+2. *func mergesort*: inner loop iterates over the array in chunks of $2 * size$, merging two adjacent chunks of size $size$ each (or less, near the end of the array).
+Each iteration of the inner loop involves merging two subarrays, and the number of such merge operations in each iteration of the outer loop is approximately $\frac{n}{2 \times size}$. This is because the inner loop is responsible for merging $\frac{n}{2 \times size}$, and each merge operation has a complexity of  $\Theta(n)$ time.
+
+3. *func inPlaceMerge*: merges two sorted subarrays, involves comparisons and rotations. The worst-case scenario for merging two subarrays of size $size$ each involves $\Theta(size)$ comparisons.
```
Aren't you doing more work than the normal merge to keep it in place?

tlimato committed:
More rigorous analysis

I was very overwhelmed and phoned in an analysis that wasn't complete last time. I took the time to properly look at every component and derive the simplified runtime analysis. I also added the references utilized for this solution like I'm doing with other repositories

tlimato committed:
Update README.md

fixed some typos and formatting

tlimato commented:
I was very overwhelmed and phoned in an analysis that wasn't complete last time. I took the time to properly look at every component and derive the simplified runtime analysis. I also added the references utilized for this solution like I'm doing with other repositories

tlimato commented:
Furthermore, I use Github copilot to assist in writing code, It's built into my vscode install I use for Proffesional work and It's helpful in accelerating projects and avoiding syntax issues. I will make this clarification on other assignments as well.
